hibernate-rx-core/src/main/java/org/hibernate/rx/event/impl/AbstractRxSaveEventListener.java

```diff
@@ -277,7 +277,8 @@ else if (identifierType == IntegerType.INSTANCE) {
 		}
 
 		if ( substitute ) {
-			persister.setPropertyValues( entity, values );
+			cascadeBeforeSave = cascadeBeforeSave
+					.thenCompose( v ->  ((RxEntityPersister) persister).setRxPropertyValues(entity, values) );
 		}
```

Comment:
@DavideD what was the reason for this, and how does it relate the change I proposed here: https://github.com/hibernate/hibernate-rx/pull/71

Reply:
When associations are involved the value is returned as a CompletionStage. I'm trying to make sure that the value returned by the CompletionStage is set before using the collection anywherelse.

Reply:
OK, I see it now. Thanks.